File: 26RemoveDuplicatesFromSortedArray.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :param nums: List[int]
        :return: int
        """
        if len(nums) == 0:
            return 0
        index = 1
        starting = 1
        length = len(nums)
        current_element = nums[0]
        count = 1
        while index <= length - 1:
            if nums[index] != current_element:
                nums[starting] = nums[index]
                current_element = nums[index]
                starting += 1
                count += 1
            else:
                pass
            index += 1
        return count




# Expected "11"
nums = []
s = Solution()
result = s.removeDuplicates(nums)
print(nums)
print(result)

File: 26RemoveDuplicatesfromSortedArray.cpp
#include <vector>
#include <iostream>

using namespace std;

class Solution
{
public:
	int removeDuplicates(vector<int>& nums)
	{
		if (nums.size() <= 1)
			return nums.size();
		int i = 0, j = 0;
		while (j != nums.size())
		{
			if (nums[i] == nums[j])
				j++;
			else
			{
				i++;
				nums[i] = nums[j];
				j++;
			}
		}
		return (i+1);
	}
};


int main()
{
	vector<int> nums = { };
	Solution sol;
	int result = sol.removeDuplicates(nums);
	cout << "The size is: " << result << endl;
	cout << "The array is listed below: \n";
	for (int k = 0; k != nums.size(); k++)
		cout << nums[k] << endl;
}